### PR TITLE
ngtcp2: SNI should not be set if host is literal IP address

### LINKED ISSUE #11827: http3: SNI should not be set if host is literal IP address
### I did this

curl --http3-only https://1.1.1.1/cdn-cgi/trace

### I expected the following

sni=off

### curl/libcurl version

curl 8.2.1 (x86_64-w64-mingw32) libcurl/8.2.1 OpenSSL/3.1.2 (Schannel) zlib/1.3 brotli/1.1.0 zstd/1.5.5 WinIDN libssh2/1.11.0 nghttp2/1.56.0 ngtcp2/0.19.1 nghttp3/0.15.0

### operating system

windows

### PR DESCRIPTION
**Pull request by vvb2060**
fix #11827

### CODE REVIEW

File: lib/vquic/curl_ngtcp2.c [hunk 1]
```diff
@@ -526,7 +528,13 @@ static CURLcode quic_init_ssl(struct Curl_cfilter *cf,
...
 
   /* set SNI */
-  SSL_set_tlsext_host_name(ctx->ssl, cf->conn->host.name);
+  if((0 == Curl_inet_pton(AF_INET, cf->conn->host.name, checkip))
+#ifdef ENABLE_IPV6
+     && (0 == Curl_inet_pton(AF_INET6, cf->conn->host.name, checkip))
+#endif
+    ) {
+    SSL_set_tlsext_host_name(ctx->ssl, cf->conn->host.name);
+  }
```
Comment (bagder) [inline lib/vquic/curl_ngtcp2.c:537]: I think it should do the SNI lowercase dance the way we do it for regular OpenSSL:

https://github.com/curl/curl/blob/39c883560a3b9ac7ddbec78f7a6a8cc13930f219/lib/vtls/openssl.c#L3765-L3769

Comment (vvb2060) [reply lib/vquic/curl_ngtcp2.c:540]: done

**Comment by jay** [comment]
curl_quiche has the same issue
https://github.com/curl/curl/blob/39c883560a3b9ac7ddbec78f7a6a8cc13930f219/lib/vquic/curl_quiche.c#L202

**Comment by vvb2060** [comment]
Sorry, I can't test other backends.  This patch only tested boringssl + ngtcp2 on Android.

**Comment by jay** [comment]
Thanks

**Pull request closed without merging**